src/runner/kafka.ts

```diff
@@ -192,31 +184,48 @@ class KafkaRunner extends BaseRunner
     }
   };
 
+  /**
+   * Important to note that Kafka is different to SQS and Redis Runners
+   * as with they are both polling consumers
+   * no timeout behaviour, need to hook into the stream provided by node-rdkafka
+   */
   process(topics: Array<string>) {
     return new Promise<KafkaConsumer>((resolve, reject) => {
       const timeoutId = setTimeout(() => {
         this.logger.error('Connection timed out');
         reject();
-      }, (this.config as KafkaConfiguration).connectionTimeout!);
+      }, this.config.connectionTimeout);
+
+      setInterval(() => {
+        if (this.paused) {
+          this.consumer.pause(this.consumer.assignments());
+        } else {
+          this.consumer.resume(this.consumer.assignments());
```

Comment:
Should we be checking the current state of the consumer in here at all? Not sure what the behaviour would be on multiple calls to resume/pause